contextualized/regression/lightning_modules.py

```diff
@@ -203,19 +217,51 @@ def _predict_y(self, C, X, beta_hat, mu_hat):
             Y = Y + self.base_y_predictor.predict_y(C, X)
         return Y
 
-    def _dataloader(self, C, X, Y, dataset_constructor, **kwargs):
+    def _dataclass_to_dataloader(self, dc):
+        """
+        :param dc: the dataclass to convert to dataloader
+        
+        """
+        dataloader = None
+
+        if type(dc) == DataLoader:
+            dataloader = dc
+        elif type(dc) in [pl.LightningDataModule, RegressionDataModule]:
+            dataloader = dc.predict_dataloader()
```

Comment:
Another place where it's unlikely that we would only ever want predictions from a specific set in the datamodule.